src/modal/__tests__/analytics.test.tsx

```diff
@@ -0,0 +1,244 @@
+      expect.objectContaining({
+        funnelType: 'modal',
+        totalFunnelSteps: 1,
+        optionalStepNumbers: [],
+        funnelName: 'Modal title',
+        stepConfiguration: [
+          {
+            isOptional: false,
+            name: 'Modal title',
+            number: 1,
+          },
+        ],
+        funnelNameSelector: expect.any(String),
+        funnelVersion: expect.any(String),
+        componentVersion: expect.any(String),
+        componentTheme: expect.any(String),
+      })
+    );
+
+    expect(FunnelMetrics.funnelStepStart).toHaveBeenCalledTimes(1);
+    expect(FunnelMetrics.funnelStepStart).toHaveBeenCalledWith(
+      expect.objectContaining({
+        stepNumber: 1,
+        stepName: 'Modal title',
+        subStepConfiguration: [
+          {
+            name: '',
+            number: 1,
+          },
+        ],
+        totalSubSteps: 1,
+        funnelInteractionId: expect.any(String),
+        stepNameSelector: expect.any(String),
+        subStepAllSelector: expect.any(String),
+      })
+    );
+  });
+
+  test('sends funnelCancelled when modal is dismissed', () => {
+    const { rerender } = render(<Modal header="Modal title" visible={true} />);
+    act(() => void jest.runAllTimers());
+    expect(FunnelMetrics.funnelStart).toHaveBeenCalledTimes(1);
+
+    rerender(<Modal header="Modal title" visible={false} />);
+    act(() => void jest.runAllTimers());
+    expect(FunnelMetrics.funnelCancelled).toHaveBeenCalledTimes(1);
+  });
+
+  test('sends funnelCancelled when modal is unmounted', () => {
+    const { rerender } = render(<Modal header="Modal title" visible={true} />);
+    act(() => void jest.runAllTimers());
+    expect(FunnelMetrics.funnelStart).toHaveBeenCalledTimes(1);
+
+    rerender(<></>);
+    act(() => void jest.runAllTimers());
+    expect(FunnelMetrics.funnelCancelled).toHaveBeenCalledTimes(1);
+  });
+
+  test('sends funnelComplete when primary button in footer is clicked', () => {
+    const modalRoot = document.createElement('div');
+    document.body.appendChild(modalRoot);
+    const { container } = render(
+      <Modal
+        header="Modal title"
+        footer={
+          <Button data-testid="submit" variant="primary">
+            Submit
+          </Button>
+        }
+        visible={true}
+        modalRoot={modalRoot}
+      />,
+      {
+        container: modalRoot,
+      }
+    );
+
+    const wrapper = createWrapper(container);
+    wrapper.findModal()!.findFooter()!.findButton('[data-testid="submit"]')?.click();
+    act(() => void jest.runAllTimers());
+    expect(FunnelMetrics.funnelComplete).toHaveBeenCalledTimes(1);
+    expect(FunnelMetrics.funnelComplete).toHaveBeenCalledWith(
+      expect.objectContaining({
+        funnelInteractionId: expect.any(String),
+      })
+    );
+  });
+
+  test('sends funnelSuccessful when primary button in footer is clicked and modal is dismissed', () => {
```

Comment:
We could potentially also test that the same event is emitted when the primary button is clicked and the modal is _unmounted_